src/main/kotlin/org/koitharu/kotatsu/parsers/site/wpcomics/vi/DocTruyen3Q.kt

```diff
@@ -1,61 +1,106 @@
 	WpComicsParser(context, MangaParserSource.DOCTRUYEN3Q, "doctruyen3qw.pro", 36) {
+	override val configKeyDomain: ConfigKey.Domain = ConfigKey.Domain(
+		"doctruyen3qw.pro", "doctruyen3qvip.com", "doctruyen3q.link",
+	)
 
-	override val userAgentKey = ConfigKey.UserAgent(UserAgents.CHROME_DESKTOP)
+	override val datePattern = "dd/MM/yyyy"
 
-	override fun onCreateConfig(keys: MutableCollection<ConfigKey<*>>) {
-		super.onCreateConfig(keys)
-		keys.add(userAgentKey)
-	}
+    override val availableSortOrders: Set<SortOrder> = EnumSet.of(
+        SortOrder.UPDATED,
+        SortOrder.POPULARITY,
+    )
 
-	override val datePattern = "dd/MM/yyyy"
+    override val filterCapabilities: MangaListFilterCapabilities
+        get() = MangaListFilterCapabilities(
+            isSearchSupported = true,
+            isTagsExclusionSupported = true,
+            isMultipleTagsSupported = false,
+        )
+
+    override suspend fun getFilterOptions(): MangaListFilterOptions {
+        return MangaListFilterOptions(
+            availableTags = availableTags(),
+            availableStates = EnumSet.of(MangaState.ONGOING, MangaState.FINISHED)
+        )
+    }
 
 	override suspend fun getListPage(page: Int, order: SortOrder, filter: MangaListFilter): List<Manga> {
 		val url = buildString {
 			append("https://")
 			append(domain)
 			append("/tim-truyen")
+			
+			if (filter.tags.isNotEmpty()) {
+				append("/")
+				append(filter.tags.first().key)
+			}
+			
+			val params = mutableListOf<String>()
+			
+			when (order) {
+				SortOrder.UPDATED -> params.add("sort=1")
+				SortOrder.POPULARITY -> params.add("sort=2")
+				else -> {} 
+			}
+			
+			filter.states.oneOrThrowIfMany()?.let { state ->
+				when (state) {
+					MangaState.ONGOING -> params.add("status=0")
+					MangaState.FINISHED -> params.add("status=1")
+					else -> params.add("status=2")
+				}
+			}
+			
 			if (filter.query?.isNotEmpty() == true) {
-				append("?keyword=")
-				append(filter.query.urlEncoded())
+				params.add("keyword=${filter.query.urlEncoded()}")
 			}
+			
 			if (page > 1) {
-				append(if (filter.query?.isNotEmpty() == true) "&" else "?")
-				append("page=")
-				append(page.toString())
+				params.add("page=$page")
+			}
+			
+			if (params.isNotEmpty()) {
+				append("?")
+				append(params.joinToString("&"))
 			}
 		}
+		
 		val doc = webClient.httpGet(url).parseHtml()
 		return parseMangaList(doc)
 	}
 
 	private fun parseMangaList(doc: Document): List<Manga> {
-		return doc.select("div.list-story-item").map { div ->
-			val href = div.selectFirstOrThrow("h3.title-book a").attrAsRelativeUrl("href")
+		return doc.select("div.item-manga").mapNotNull { div ->
+			val href = div.selectFirst("h3 a")?.attrAsRelativeUrl("href") ?: return@mapNotNull null
+			val tagElements = div.select("p.info a[href*=tim-truyen]")
+			val tags = tagElements.mapNotNullToSet { a ->
+				val key = a.attr("href").substringAfterLast("/")
+				availableTags().find { it.key == key }
```

Comment:
It would be better to stor `availableTags` value into a variable out of the loop